src/block/carousel/frontend-carousel.js

```diff
@@ -166,20 +191,56 @@ class _StackableCarousel {
 
 	nextSlide = () => {
 		let newSlide = this.currentSlide + 1
-		if ( newSlide > this.maxSlides() ) {
+
+		if ( this.infiniteScroll && newSlide > this.slideEls.length ) {
+			this.swapSlides( 'N' )
+			return
+		}
+
+		if ( ! this.infiniteScroll && newSlide > this.maxSlides() ) {
 			newSlide = this.slideOffset
 		}
 		this.goToSlide( newSlide )
 	}
 
 	prevSlide = () => {
 		let newSlide = this.currentSlide - 1
+
+		if ( this.infiniteScroll && newSlide < this.slideOffset ) {
+			this.swapSlides( 'P' )
+			return
+		}
+
 		if ( newSlide < this.slideOffset ) {
```

Comment:
Seeing the `if` condition in the `nextSlide`, would you also need `! this.infiniteScroll` in here?
```suggestion
		if ( ! this.infiniteScroll && newSlide < this.slideOffset ) {
```